src/composeapp.cc

```diff
@@ -42,6 +44,68 @@ void ComposeApp::remove() {
   }
 }
 
+bool ComposeApp::check() {
+  bool result = false;
+
+  do {
+    bool cmd_res{false};
+    std::string cmd_output;
+
+    std::tie(cmd_res, cmd_output) = cmd("config");
+    if (!cmd_res) {
+      LOG_WARNING << "Missing and incomplete App config: " << name_;
+      break;
+    }
+
+    // calculate a number of container images that a given App consists of
+    std::string::size_type find_pos{0};
+    int expected_image_number{0};
+
+    while ((find_pos = cmd_output.find("image:", find_pos)) != std::string::npos) {
+      ++expected_image_number;
+      ++find_pos;
+    }
+
+    // Get a number of installed images
+    std::tie(cmd_res, cmd_output) = cmd("images -q");
+    if (!cmd_res) {
+      LOG_WARNING << "Failed to get a list of App's images: " << name_;
+      break;
+    }
+
+    int installed_image_number =
+        std::count_if(cmd_output.begin(), cmd_output.end(), [](const char& symbol) { return symbol == '\n'; });
+
+    if (installed_image_number < expected_image_number) {
+      LOG_DEBUG << "Number of installed images is less than a number of images specified in the compose file"
+                << "; App: " << name_ << "; expected image number: " << expected_image_number
+                << "; number of installed images: " << installed_image_number;
+      break;
+    }
+
+    // Get a number of running container images
```

Comment:
a bit of a premature optimization. but the docker-compose binary is a little slow to "start". Checking if the image exists and is running is great for debugging, but we really only need this "is it running" check right?

Reply:
an old App can be running while an update of a new App was interrupted, although in this case Target in the DB should not be updated so the next update cycle should start App update process anyway... I will check it out.

Reply:
removed 'images' check.